client/lxd_instances.go

```diff
@@ -211,6 +212,169 @@ func (r *ProtocolLXD) UpdateInstances(state api.InstancesPut, ETag string) (Oper
 	return op, nil
 }
 
+func (r *ProtocolLXD) rebuildInstance(instance api.InstancesPost) (Operation, error) {
+	path, _, err := r.instanceTypeToPath(instance.Type)
+	if err != nil {
+		return nil, err
+	}
+
+	// Send the request
+	op, _, err := r.queryOperation("POST", fmt.Sprintf("%s/%s/rebuild?project=%s", path, url.PathEscape(instance.Name), r.project), instance, "")
+	if err != nil {
+		return nil, err
+	}
+
+	return op, nil
+}
+
+func (r *ProtocolLXD) tryRebuildInstance(req api.InstancesPost, urls []string, op Operation) (RemoteOperation, error) {
+	reflReq := reflect.ValueOf(&req).Elem()
```

Comment:
We shouldn't need this complexity. There should only need to be one source field for this type of request.